services/blockstorage/sync/block_sync_test.go

```diff
@@ -1,48 +1,21 @@
 package sync
 
 import (
+	"github.com/orbs-network/orbs-network-go/synchronization"
 	"github.com/stretchr/testify/require"
 	"testing"
-	"time"
 )
 
 func TestBlockSyncStartsWithImmediateSync(t *testing.T) {
-	h := newBlockSyncHarness().withNoCommitTimeout(time.Hour) // we want to see that the sync immediately starts, and not in an hour
+	manualNoCommitTimer := synchronization.NewTimerWithManualTick()
+	h := newBlockSyncHarnessWithManualNoCommitTimeoutTimer(manualNoCommitTimer)
 
 	h.expectingSyncOnStart()
 
-	sync := NewBlockSync(h.ctx, h.config, h.gossip, h.storage, h.logger, h.m)
+	sync := NewBlockSync(h.ctx, h.config, h.gossip, h.storage, h.logger, h.metricFactory)
 
 	h.eventuallyVerifyMocks(t, 2) // just need to verify we used gossip/storage for sync
 	h.cancel()
 	shutdown := h.waitForShutdown(sync)
 	require.True(t, shutdown, "expecting state to be set to nil (=shutdown)")
 }
-
-func TestBlockSyncStaysInIdleOnBlockCommitExternalMessage(t *testing.T) {
```

Comment:
this was the only real flaky test, where is this logic tested now? specifically this tested that `HandleBlockCommitted` works

Reply:
I'lll try to add it as an external call in the unit test